R/seroprevalence_data.R

```diff
@@ -139,6 +139,31 @@ prepare_bin_data <- function(serodata) {
   return(xx)
 }
 
+#' Returns the probability of being seropositive for age-varying
+#' force-of-infection including seroreversion
+#'
+#' @param age Integer corresponding to the age of the exposed cohort
+#' @param foi Numeric atomic vector corresponding to the age-varying FoI to
+#' simulate from
+#' @param mu Seroreversion rate
+#' @return probability of being seropositive for age-varying FoI
+#' including seroreversion
+probability_exact_age_varying <- function(
+    age,
+    foi,
+    mu = 0
+) {
+  probability <- 0
+  # solves ODE exactly within pieces
+  for (i in 1:age) {
+    probability <-
+      (1 / (foi[i] + mu)) * exp(- (foi[i] + mu)) *
+      (foi[i] * (exp(foi[i] + mu)  - 1) + probability * (foi[i] + mu)
+      )
```

Comment:
I'm a bit confused -- have you changed this to the new version (with only a single exponential)?